src/Microsoft.OData.Core/ODataBatchOperationRequestMessage.cs

```diff
@@ -55,17 +64,19 @@ public sealed class ODataBatchOperationRequestMessage : IODataRequestMessage, IO
             IODataBatchOperationListener operationListener,
             string contentId,
             IODataUrlResolver urlResolver,
-            bool writing)
+            bool writing,
+            IList<string> dependsOnRequestIds)

```

Comment:
>IList<string> [](start = 12, length = 13)

We typically pass as IEnumerable, rather than List.  I would make this IEnumerable<string>, and in the constructor set dependsOnRequestIds=new List<string>(dependsOnRequestIds)